deep_quoridor/src/agents/simple.py

```diff
@@ -1,61 +1,175 @@
 import random
+from dataclasses import dataclass, fields
+from typing import Optional
 
-from quoridor import ActionEncoder, Player, Quoridor, construct_game_from_observation
-from quoridor_env import StepRewardCalculator
+import numpy as np
+from quoridor import Action, ActionEncoder, Player, Quoridor, WallAction, construct_game_from_observation
+from utils import SubargsBase
 
 from agents.core import Agent
 
+# We use a large reward to encourage the agent to win the game, but we can't use infinity
+# because then multiplying by a discount factor won't decrease the reward.
+WINNING_REWARD = 1e6
 
-def sample_random_action_sequence(game: Quoridor, player: Player, opponent: Player, max_path_length: int):
+
+@dataclass
+class SimpleParams(SubargsBase):
+    nick: Optional[str] = None
+
+    # How many moves to look ahead in the minimax algorithm.
+    max_depth: int = 2
+
+    # How many actions to consider at each minimax stage.
+    branching_factor: int = 20
+
+    # We use a Guassian distribution centered around each player's position to sample wall actions.
+    # Sigma is the standard deviation of this distribution; a smaller sigma means the agent is more
+    # likely to place walls near themselves or their opponent, rather than somewhere else on the board.
+    wall_sigma: float = 0.5
+
+    # Future rewards are worth less. This makes the agent try to win quickly. Without this,
+    # once the agent knows it can win, it may oscillate between two moves instead of just winning.
+    discount_factor: float = 0.99
+
+
+def compute_wall_weight(wall: WallAction, position_1: np.ndarray, position_2: np.ndarray, sigma) -> float:
```

Comment:
If in the profiler this is taking a significant amount of time, it may be worth vectorizing it, i.e. passing all the wall actions at once, since all those math operations would be much more efficient

Reply:
Also, I just tried doing:
```
    print(min_distance, weight)
```

and got for example:
```
1.0 0.1353352832366127
1.4142135623730951 0.018315638888734165
2.23606797749979 4.539992976248477e-05
```
So, this decreases very quickly, the 3rd row is a difference in 1 vertical and 2 horizontal and it's already 3 orders of magnitude less likely to be chosen than one that is just 1 in difference, so almost never would be chosen.

Additionally, I realized the following: the coordinate of the wall is the top left, so for example a wall sitting right below the pawn would have different distances if it's extending to the right or to the left.  I think we should consider the distance to the center of the wall

Reply:
Yes on both points. The value decreasing quickly was (mostly) intentional since I wanted to really see the effect. Since it chooses wall actions without replacement, this results in almost always choosing the N closest walls. 

The distance being wrong is a bigger bug; I'll take a look at fixing that today.

Reply:
Updated to use the distance from the center of the wall to the player position, and performance is better now!

I did try increasing wall_sigma from 0.5 to 1, but performance slightly decreased, so I'll leave it at 0.5 for now.